# Introduction
This case study is my Capstone Project for Google Data Analystics Professional Certificate. To complete this case study I used the 6 steps of Data analysis to present my findings.

Title: Bellabeat Case Study

by: John Riggs

# Step 1: Ask

**Background**

Since 2013, Bellabeat has been creating advanced technology products specifically designed to improve women's health. Their smart products are not only aesthetically pleasing but also offer women the knowledge and tools to take control of their health and habits. The company has grown quickly and established itself as a technology-driven wellness brand for women. 

According to Urška Sršen, the co-founder and Chief Creative Officer, analyzing data from non-Bellabeat products like FitBit fitness trackers could uncover further opportunities for growth.

**Task**

Analyze the FitBit Fitness Tracker data to gain insight on how consumers are using this product and analyze any trends/insights for Beallabeat's marketing team.

**Objectives**
1. What are some trends in smart device usage?
2. How could these trends apply to Bellabeat customers?
3. How could these trends help influence Bellabeat marketing strategy?

**Produce**
1. A clear summary of the business task
2. A description of all data sources used
3. Documentation of any cleaning or manipulation of data
4. A summary of your analysis
5. Supporting visualizations and key findings
6. Your top high-level content recommendations based on your analysis

**Stakeholders**

Urška Sršen: Bellabeat’s cofounder and Chief Creative Officer
    
Sando Mur: Mathematician, Bellabeat’s cofounder and key member of the Bellabeat executive team
    
Bellabeat marketing analytics team: A team of data analysts guiding Bellabeat's marketing strategy.





# Step 2: Prepare

**Information on Data Source**


# Step 3: Process

For this study I decided to use Python to prepare and process all of the data.

**3.1 Preparing the Enviornment**

For this case study the packages being used are: *pandas*, *altair*, *datetime*, and *vega_datasets*

In [2]:
import pandas as pd
import altair as alt
import datetime as dt
from vega_datasets import data 

**3.2 Importing the data set**

In [3]:
# read_csv allows us to read the required CSV file
daily_activity = pd.read_csv("Data/dailyActivity_merged.csv")

**3.3 Data cleaning and manipulation**

From here we will do 3 important steps.
    1. Observe the data and Familiarize ourselves with it.
    2. Check for any missing, null, or NaN values.
    3. Check the data

Using the .head() function allows use to see our column names and the rows to familirize ourselves.

In [4]:
daily_activity.head()

Id ActivityDate  TotalSteps  TotalDistance  TrackerDistance  \
0  1503960366    4/12/2016       13162           8.50             8.50   
1  1503960366    4/13/2016       10735           6.97             6.97   
2  1503960366    4/14/2016       10460           6.74             6.74   
3  1503960366    4/15/2016        9762           6.28             6.28   
4  1503960366    4/16/2016       12669           8.16             8.16   

   LoggedActivitiesDistance  VeryActiveDistance  ModeratelyActiveDistance  \
0                       0.0                1.88                      0.55   
1                       0.0                1.57                      0.69   
2                       0.0                2.44                      0.40   
3                       0.0                2.14                      1.26   
4                       0.0                2.71                      0.41   

   LightActiveDistance  SedentaryActiveDistance  VeryActiveMinutes  \
0                 6.06                      0.0                 25   
1                 4.71                      0.0                 21   
2                 3.91                      0.0                 30   
3                 2.83                      0.0                 29   
4                 5.04                      0.0                 36   

   FairlyActiveMinutes  LightlyActiveMinutes  SedentaryMinutes  Calories  
0                   13                   328               728      1985  
1                   19                   217               776      1797  
2                   11                   181              1218      1776  
3                   34                   209               726      1745  
4                   10                   221               773      1863

Finding any missing values

In [5]:
non_values = daily_activity.isnull().sum()
non_values[:]

Id                          0
ActivityDate                0
TotalSteps                  0
TotalDistance               0
TrackerDistance             0
LoggedActivitiesDistance    0
VeryActiveDistance          0
ModeratelyActiveDistance    0
LightActiveDistance         0
SedentaryActiveDistance     0
VeryActiveMinutes           0
FairlyActiveMinutes         0
LightlyActiveMinutes        0
SedentaryMinutes            0
Calories                    0
dtype: int64

In [6]:
daily_activity.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 940 entries, 0 to 939
Data columns (total 15 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Id                        940 non-null    int64  
 1   ActivityDate              940 non-null    object 
 2   TotalSteps                940 non-null    int64  
 3   TotalDistance             940 non-null    float64
 4   TrackerDistance           940 non-null    float64
 5   LoggedActivitiesDistance  940 non-null    float64
 6   VeryActiveDistance        940 non-null    float64
 7   ModeratelyActiveDistance  940 non-null    float64
 8   LightActiveDistance       940 non-null    float64
 9   SedentaryActiveDistance   940 non-null    float64
 10  VeryActiveMinutes         940 non-null    int64  
 11  FairlyActiveMinutes       940 non-null    int64  
 12  LightlyActiveMinutes      940 non-null    int64  
 13  SedentaryMinutes          940 non-null    int64  
 14  Calories  

Finding any unique IDs for the data.

In [7]:
unique_values = len(pd.unique(daily_activity["Id"]))
print("# of Unique ID: " + str(unique_values))

# of Unique ID: 33


The above data shows up there there are not any nulls, missing values, or typos. The number of rows and columns and what type of dtype each column is.

The next steps were converting the ActivityDate to datatime64 dtype instead of an object and creating a new column to give us what day of the week each activity date was on.

We will also create a new column of Total Exercise Minutes and Total Exercise Hours

In [8]:
#converting to datetime 
daily_activity["ActivityDate"] = pd.to_datetime(daily_activity['ActivityDate'], format ="%m/%d/%Y")
daily_activity.head()


Id ActivityDate  TotalSteps  TotalDistance  TrackerDistance  \
0  1503960366   2016-04-12       13162           8.50             8.50   
1  1503960366   2016-04-13       10735           6.97             6.97   
2  1503960366   2016-04-14       10460           6.74             6.74   
3  1503960366   2016-04-15        9762           6.28             6.28   
4  1503960366   2016-04-16       12669           8.16             8.16   

   LoggedActivitiesDistance  VeryActiveDistance  ModeratelyActiveDistance  \
0                       0.0                1.88                      0.55   
1                       0.0                1.57                      0.69   
2                       0.0                2.44                      0.40   
3                       0.0                2.14                      1.26   
4                       0.0                2.71                      0.41   

   LightActiveDistance  SedentaryActiveDistance  VeryActiveMinutes  \
0                 6.06                      0.0                 25   
1                 4.71                      0.0                 21   
2                 3.91                      0.0                 30   
3                 2.83                      0.0                 29   
4                 5.04                      0.0                 36   

   FairlyActiveMinutes  LightlyActiveMinutes  SedentaryMinutes  Calories  
0                   13                   328               728      1985  
1                   19                   217               776      1797  
2                   11                   181              1218      1776  
3                   34                   209               726      1745  
4                   10                   221               773      1863

In [9]:
cols2 = ['Id', 'ActivityDate', 'DayOfTheWeek', 'TotalSteps', 'TotalDistance', 'TrackerDistance', 'LoggedActivitiesDistance', 
'VeryActiveDistance', 'ModeratelyActiveDistance', 'LightActiveDistance', 'SedentaryActiveDistance', 'VeryActiveMinutes', 
'FairlyActiveMinutes', 'LightlyActiveMinutes', 'SedentaryMinutes', 'TotalExerciseMinutes', 'TotalExerciseHours', 'Calories']

# da = daily activity 
# Chaning the variable to make it easier
da = daily_activity.reindex(columns=cols2)

da["DayOfTheWeek"] = da["ActivityDate"].dt.day_name()

da["DayOfTheWeek"].head()

0      Tuesday
1    Wednesday
2     Thursday
3       Friday
4     Saturday
Name: DayOfTheWeek, dtype: object

In [10]:
da.rename(columns = {"Id":"Id", "ActivityDate":"Date", "DayOfTheWeek":"DayOfTheWeek", 
"TotalSteps":"TotalSteps", "TotalDistance":"TotalDistance", 
"TrackerDistance":"TrackerDistance", "LoggedActivitiesDistance":"LoggedActivitiesDistance", 
"VeryActiveDistance":"VeryActiveDistance", "ModeratelyActiveDistance":"ModeratelyActiveDistance", 
"LightActiveDistance":"LightActiveDistance", "SedentaryActiveDistance":"SedentaryActiveDistance", 
"VeryActiveMinutes":"VeryActiveMinutes", "FairlyActiveMinutes":"FairlyActiveMinutes", 
"LightlyActiveMinutes":"LightlyActiveMinutes", "SedentaryMinutes":"SedentaryMinutes", 
"TotalExerciseMinutes":"TotalExerciseMinutes","TotalExerciseHours":"TotalExerciseHours",
"Calories":"Calories"}, inplace = True)

print(da.columns.values)
da.head()

['Id' 'Date' 'DayOfTheWeek' 'TotalSteps' 'TotalDistance' 'TrackerDistance'
 'LoggedActivitiesDistance' 'VeryActiveDistance'
 'ModeratelyActiveDistance' 'LightActiveDistance'
 'SedentaryActiveDistance' 'VeryActiveMinutes' 'FairlyActiveMinutes'
 'LightlyActiveMinutes' 'SedentaryMinutes' 'TotalExerciseMinutes'
 'TotalExerciseHours' 'Calories']


Id       Date DayOfTheWeek  TotalSteps  TotalDistance  \
0  1503960366 2016-04-12      Tuesday       13162           8.50   
1  1503960366 2016-04-13    Wednesday       10735           6.97   
2  1503960366 2016-04-14     Thursday       10460           6.74   
3  1503960366 2016-04-15       Friday        9762           6.28   
4  1503960366 2016-04-16     Saturday       12669           8.16   

   TrackerDistance  LoggedActivitiesDistance  VeryActiveDistance  \
0             8.50                       0.0                1.88   
1             6.97                       0.0                1.57   
2             6.74                       0.0                2.44   
3             6.28                       0.0                2.14   
4             8.16                       0.0                2.71   

   ModeratelyActiveDistance  LightActiveDistance  SedentaryActiveDistance  \
0                      0.55                 6.06                      0.0   
1                      0.69                 4.71                      0.0   
2                      0.40                 3.91                      0.0   
3                      1.26                 2.83                      0.0   
4                      0.41                 5.04                      0.0   

   VeryActiveMinutes  FairlyActiveMinutes  LightlyActiveMinutes  \
0                 25                   13                   328   
1                 21                   19                   217   
2                 30                   11                   181   
3                 29                   34                   209   
4                 36                   10                   221   

   SedentaryMinutes  TotalExerciseMinutes  TotalExerciseHours  Calories  
0               728                   NaN                 NaN      1985  
1               776                   NaN                 NaN      1797  
2              1218                   NaN                 NaN      1776  
3               726                   NaN                 NaN      1745  
4               773                   NaN                 NaN      1863

In [11]:
# This takes all of the min and sums them up
da['TotalExerciseMinutes'] = da['VeryActiveMinutes'] + da['FairlyActiveMinutes'] + da['LightlyActiveMinutes'] + da['SedentaryMinutes'] 

# This will take the minutes and convert them to hours
da["TotalExerciseHours"] = round(da["TotalExerciseMinutes"] / 60)
da["TotalExerciseHours"].head(5)

0    18.0
1    17.0
2    24.0
3    17.0
4    17.0
Name: TotalExerciseHours, dtype: float64

The Data and manipulation of data is now completed.

# Step 4: Analyzing the data

**4.1 Performing the Calculations**

Pulling the statstics with da for analysis


In [12]:
da.describe()

Id    TotalSteps  TotalDistance  TrackerDistance  \
count  9.400000e+02    940.000000     940.000000       940.000000   
mean   4.855407e+09   7637.910638       5.489702         5.475351   
std    2.424805e+09   5087.150742       3.924606         3.907276   
min    1.503960e+09      0.000000       0.000000         0.000000   
25%    2.320127e+09   3789.750000       2.620000         2.620000   
50%    4.445115e+09   7405.500000       5.245000         5.245000   
75%    6.962181e+09  10727.000000       7.712500         7.710000   
max    8.877689e+09  36019.000000      28.030001        28.030001   

       LoggedActivitiesDistance  VeryActiveDistance  ModeratelyActiveDistance  \
count                940.000000          940.000000                940.000000   
mean                   0.108171            1.502681                  0.567543   
std                    0.619897            2.658941                  0.883580   
min                    0.000000            0.000000                  0.000000   
25%                    0.000000            0.000000                  0.000000   
50%                    0.000000            0.210000                  0.240000   
75%                    0.000000            2.052500                  0.800000   
max                    4.942142           21.920000                  6.480000   

       LightActiveDistance  SedentaryActiveDistance  VeryActiveMinutes  \
count           940.000000               940.000000         940.000000   
mean              3.340819                 0.001606          21.164894   
std               2.040655                 0.007346          32.844803   
min               0.000000                 0.000000           0.000000   
25%               1.945000                 0.000000           0.000000   
50%               3.365000                 0.000000           4.000000   
75%               4.782500                 0.000000          32.000000   
max              10.710000                 0.110000         210.000000   

       FairlyActiveMinutes  LightlyActiveMinutes  SedentaryMinutes  \
count           940.000000            940.000000        940.000000   
mean             13.564894            192.812766        991.210638   
std              19.987404            109.174700        301.267437   
min               0.000000              0.000000          0.000000   
25%               0.000000            127.000000        729.750000   
50%               6.000000            199.000000       1057.500000   
75%              19.000000            264.000000       1229.500000   
max             143.000000            518.000000       1440.000000   

       TotalExerciseMinutes  TotalExerciseHours     Calories  
count            940.000000          940.000000   940.000000  
mean            1218.753191           20.313830  2303.609574  
std              265.931767            4.437283   718.166862  
min                2.000000            0.000000     0.000000  
25%              989.750000           16.000000  1828.500000  
50%             1440.000000           24.000000  2134.000000  
75%             1440.000000           24.000000  2793.250000  
max             1440.000000           24.000000  4900.000000

# Step 5: Sharing the Data

This is where we will create visualizations and communicate our findings 

In [13]:
# creating a new CSV with the updated columns and information
da.to_csv('da.csv')

**5.1 Data Visualisation and Findings**

In [14]:
df = pd.read_csv("da.csv")
hist = alt.Chart(df).mark_bar().encode(
    alt.X('DayOfTheWeek:N', axis=alt.Axis(title='Days of the Week', labelFontSize=12), 
    sort=alt.EncodingSortField(field='DayOfTheWeek', op='count', order='ascending')),
    alt.Y('count():Q', axis=alt.Axis(title='Frequency'))
).properties(width=600, height=400, title="No. of Logins During the Week")

hist

alt.Chart(...)

**Usage Frequency throughout the Week**

This histogram shows us the frequence use of the FitBit app in terms of days of the week.

1. Through our data we discovered that users are more likely to track their data Tuesday - Thursday in the middle of the week rather than the end/beginning of the week.
    
2. The data also shows a steady drop starting on Friday and continuning to drop up until Monday.

In [32]:

brush = alt.selection(type='interval')
base = alt.Chart(df).add_selection(brush)

points = base.mark_point().encode(
    x = alt.X('TotalSteps:Q', title = 'Steps Taken'),
    y = alt.Y('Calories:Q', title = 'Calories Burned'),
).properties(width=600, height=400, title="Calories Burned for Steps Taken")

rule = alt.Chart(df).mark_rule(color='blue').encode(
    x='mean(TotalSteps):Q'
)

rule2 = alt.Chart(df).mark_rule(color='red').encode(
    y='mean(Calories):Q'
)

(points + rule + rule2)


alt.LayerChart(...)

In [34]:
scatter = alt.Chart(df).mark_circle().encode(
    alt.X('TotalExerciseHours:Q', title="Hours Logged"),
    alt.Y('Calories:Q', title="Calories Burned"),
).properties(width=800, height=400, title="Calories Burned per Logged Hour")

med_steps = alt.Chart(df).mark_rule(color='blue').encode(
    x='mean(TotalExerciseHours):Q'
)

med_hours = alt.Chart(df).mark_rule(color='purple').encode(
    y='mean(Calories):Q'  
)   

med_sed = alt.Chart(df).mark_rule(color='red').encode(
    y='mean(SedentaryMinutes):Q'
)
  
(scatter + med_steps + med_sed + med_hours)

alt.LayerChart(...)

**Calories burned per Hour**

This scatter plot is showing a weak positive correlation when it comes to the increase in hours logged versus more calories being burned.

There continue to be some outliers. One being the zero values and the other being a log of 24 hours but no calories were burned.

In [17]:
lightly_active_minutes = da["LightlyActiveMinutes"].sum()
fairly_active_minutes = da["FairlyActiveMinutes"].sum()
very_active_minutes = da['VeryActiveMinutes'].sum()
sedentary_minutes = da["SedentaryMinutes"].sum()
total_min = da["TotalExerciseMinutes"].sum()

print(lightly_active_minutes / total_min * 100, fairly_active_minutes / total_min * 100, very_active_minutes / total_min * 100, sedentary_minutes / total_min * 100)

15.820493214202166 1.1130139975629088 1.7366021081886964 81.32989068004622


In [35]:
source = pd.DataFrame(
    {"category": ["Lightly Active Minutes", "Fairly Active Minutes", "Very Active Minutes", 
    "Sedentary Minutes"], "values": [15.8, 1.1, 1.7, 81.3]}
)

base = alt.Chart(source).encode(
    theta=alt.Theta("values:Q", stack=True),
    radius=alt.Radius("values", scale=alt.Scale(type="sqrt", zero=True, rangeMin=150)),
    color="category:N",
).properties(width=300, height=300)

c1 = base.mark_arc(innerRadius=15, stroke="#fff")

c2 = base.mark_text(radiusOffset=10).encode(text="values:Q")

c1 + c2


alt.LayerChart(...)

**Percentage of Activity in Minutes**

The pie chart shows that Sedenatry minutes are the largest with 81%. This shows us that the data being logged into the FitBit app are mainly daily activites such as commute, moving from one sppot to another or running errands.

The App is rarely being used to track fitness this is shown by the very active minutes being 1.7% and fairly active being 1.1%. 

This allows us to see that the FitBit app is not attracting the population it was created for. 


# Step 6: Act

This will be where we share our recommendations for the business.

**What are the trends identified?**

The majority of users are using the FitBit to track sedentary activites instead of tracking their health habits.

Users prefer to track their activites in the middle of the week compared to the end of the week. This could be do to a number of factors depending on how users like to spend their weekends. 

**How could these trends apply to Bellabeat customers?**

Since both companies main focus is providing health and fitness data and encouraging the idea of knowing their habits, these trends could be applied to Bellabeat custonmers and show similarites in customer base.

**How could these trends help influence Bellabeat marketing strategy?**

The marketing team can encourage users by suggesting simple exercises that they can do at home. They can also show the calories being burnt on the app to show progress.
    
Bellabeat app can add a notification feature to encourage users to be more active on the weekends. They can also add events or fun things that would make people excited to exercise on the weekends. 
